### PR TITLE
Add a Tenor wrapper 

### PR DESCRIPTION
**Pull request by zhichao-li**
## What changes were proposed in this pull request?
1. Rename bigdl to dl and move dev folder under the test folder
2. Add a tensor wrapper: JTensor, which can easy our work when need to pass or return Tensor to/from Scala.
This class should be use Internally to save some boilerplate code as the user should operate on ndarray directly.
3.Add CrossEntropyCriterion
## How was this patch tested?
doctest

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala [hunk 1]
```diff
@@ -99,6 +101,23 @@ class PythonBigDL[T: ClassTag](implicit ev: TensorNumeric[T]) extends Serializab
...
   }
 
+  def toTensor(jTensor: JTensor): Tensor[T] = {
+    Tensor(jTensor.storage.asScala.toArray.asInstanceOf[Array[T]],
+      jTensor.shape.asScala.toArray)
+  }
+
+  def toJTensor(tensor: Tensor[T]): JTensor = {
+    // TODO: we should clone here, in case the underlying storage large than the tensor size.
+    val cloneTensor = tensor
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: Need to do a copy

Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: clone method is break at the moment. Would add it back after fixing that issue.

Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: https://github.com/intel-analytics/BigDL/issues/641

Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: Don't use clone - you need to make sure it's continuous. use resizeAs() and copy() instead.

Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: clone would copy and create a new tensor for us, what it really does is exactly combining resizeAs() and copy().

**Pull request merged**